Source/Events.Filters/Internal/AbstractFilterProcessor.cs

```diff
@@ -57,13 +58,20 @@ protected override RetryProcessingState GetRetryProcessingState(FilterEventReque
         /// <inheritdoc/>
         protected override Task<TFilterResponse> Handle(FilterEventRequest request, CancellationToken cancellationToken)
         {
-            var committed = _converter.ToSDK(request.Event);
-            if (committed.Event is TEventType typedEvent)
+            try
             {
-                return Filter(typedEvent, committed.DeriveContext());
-            }
+                var committed = _converter.ToSDK(request.Event);
+                if (committed.Event is TEventType typedEvent)
+                {
+                    return Filter(typedEvent, committed.DeriveContext());
+                }
 
-            throw new EventTypeIsIncorrectForFilter(typeof(TEventType), committed.Event.GetType());
+                throw new EventTypeIsIncorrectForFilter(typeof(TEventType), committed.Event.GetType());
+            }
+            catch (CouldNotDeserializeEvent ex)
+            {
+                throw new CouldNotDeserializeEventFromScope(request.ScopeId.ToGuid(), ex);
```

Comment:
Should pass in the actual ScopeId Concept
```suggestion
                throw new CouldNotDeserializeEventFromScope(request.ScopeId.To<ScopeId>(), ex);
```